template.yaml

```diff
@@ -233,3 +243,78 @@ Resources:
           Value: !Sub "${AWS::StackName}-LoggingKmsKey"
         - Key: Stack
           Value: !Sub "${AWS::StackName}"
+
+  AuditEventQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !Ref AuditEventQueueEncryptionKeyAlias
+      VisibilityTimeout: 70
+      MessageRetentionPeriod: 1209600
+      RedriveAllowPolicy:
+        redrivePermission: denyAll
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt AuditEventDeadLetterQueue.Arn
+        maxReceiveCount: 10
+
+  AuditEventDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !Ref AuditEventQueueEncryptionKeyAlias
+
+  AuditEventQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref AuditEventQueue
+      PolicyDocument:
+        Statement:
+          - Sid: "AllowReadByTxMAAccount"
+            Effect: Allow
+            Action:
+              - "sqs:ChangeMessageVisibility"
+              - "sqs:DeleteMessage"
+              - "sqs:GetQueueAttributes"
+              - "sqs:ReceiveMessage"
+            Resource: !GetAtt AuditEventQueue.Arn
+            Principal:
+              AWS: !FindInMap [ EnvironmentMap, !Ref Environment, txmaAccountId ]
+
+  AuditEventQueueEncryptionKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Symmetric key used to encrypt audit messages at rest in SQS
+      EnableKeyRotation: true
+      KeySpec: SYMMETRIC_DEFAULT
+      KeyPolicy:
+        Version: 2012-10-17
+        Statement:
+          - Sid: Enable Root access
+            Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - "kms:*"
+            Resource: '*'
+          - Sid: Allow decryption of events by TxMA
+            Effect: Allow
+            Principal:
+              AWS: !FindInMap [ EnvironmentMap, !Ref Environment, txmaAccountId ]
+            Action:
+              - "kms:decrypt"
+            Resource: '*'
+      Tags:
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Name
+          Value: !Sub "${AWS::StackName}-AuditEventQueueEncryptionKey"
+        - Key: Stack
+          Value: !Sub "${AWS::StackName}"
+
+  AuditEventQueueEncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !If
+        - IsDev
+        - !Sub "alias/${AWS::StackName}/auditEventQueueEncryptionKey"
```

Comment:
This special treatment of dev/local is different to the way we do things in other repos. I assume this is intentional and maybe something we overlooked in SPOT/CIMIT that we might choose to address in the future?